pipeline/runtime/runtestsV2.go

```diff
@@ -0,0 +1,330 @@
+		RunOnlySelectedTests = false
+	} else if selection.SelectAll {
+		log.Infoln("Test Intelligence determined to run all the tests")
+		RunOnlySelectedTests = false // TI selected all the tests to be run
+	} else {
+		log.Infoln(fmt.Sprintf("Running tests selected by Test Intelligence: %s", selection.Tests))
+	}
+
+	return selection, true
+}
+
+func createJavaConfigFile(tmpDir string, fs filesystem.FileSystem, log *logrus.Logger) (string, error) {
+
+	dir := fmt.Sprintf(outDir, tmpDir)
+	err := fs.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", dir))
+		return "", err
+	}
+
+	iniFileDir := fmt.Sprintf("%s/new", tmpDir)
+	err = fs.MkdirAll(iniFileDir, os.ModePerm)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create nested directory %s", iniFileDir))
+		return "", err
+	}
+	iniFile := fmt.Sprintf("%s/config.ini", iniFileDir)
+	data := fmt.Sprintf(`outDir: %s
+	logLevel: 0 
+	logConsole: false
+	writeTo: JSON
+	packageInference: true`, dir)
+
+	log.Infof("Attempting to write to %s with config:\n%s", iniFile, data)
+	f, err := fs.Create(iniFile)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create config file %s", iniFile))
+		return "", err
+	}
+
+	_, err = f.WriteString(data)
+	defer f.Close()
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not write %s to config file %s", data, iniFile))
+		return "", err
+	}
+
+	return iniFile, nil //path of config.ini file
+}
+
+func getPreCmd(tmpFilePath string, fs filesystem.FileSystem, log *logrus.Logger) (string, error) {
+	var preCmd string
+	iniFilePath, err := createJavaConfigFile(tmpFilePath, fs, log)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not create java agent config file in path %s", iniFilePath))
+		return "", err
+	}
+
+	err = writetoBazelrcFile(iniFilePath, log, fs, tmpFilePath)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("failed to write in .bazelrc file"))
+		return "", err
+	}
+	javaAgentPath := fmt.Sprintf("%s%s%s", tmpFilePath, javaNewAgentPath, javaNewAgentJar)
+	agentArg := fmt.Sprintf(javaNewAgentArg, javaAgentPath, iniFilePath)
+	preCmd = fmt.Sprintf("export JAVA_TOOL_OPTIONS=%s", agentArg)
+	return preCmd, nil
+}
+
+func downloadJavaAgent(ctx context.Context, path string, fs filesystem.FileSystem, log *logrus.Logger) error {
+
+	javaAgentPath := fmt.Sprintf("%s%s", javaNewAgentPath, javaNewAgentJar)
+	dir := filepath.Join(path, javaAgentPath)
+	err := utils.DownloadFile(ctx, dir, javaNewAgentUrl, fs)
+	if err != nil {
+		log.WithError(err).Errorln(fmt.Sprintf("could not download java agent"))
+		return err
+	}
+	return nil
+}
+
+func createFilterFile(ctx context.Context, fs filesystem.FileSystem, stepID, workspace string, log *logrus.Logger,
+	tiConfig *tiCfg.Cfg, path string) error {
+
+	isManualExecution := utils.IsManualExecution(tiConfig)
```

Comment:
@anurag-harness how do we know if we do not want to attach instrumentation? We don't have `runOnlySelectedTests` as task input.

Reply:
Yeah we are always attaching instrumentation